snowshu/adapters/target_adapters/base_target_adapter.py

```diff
@@ -152,156 +105,90 @@ def load_data_into_relation(self, relation: Relation, data: pd.DataFrame) -> Non
-            Should be defined in specific target adapters due to different setup of different dbs
-
-        Args:
-            source_adapter_name: the classname of the source adapter
-            incremental_image: the name of incremental image to initialize,
-                if specified will override default image
+    def _get_data_type(self, source_type: str) -> DataType:
         """
-
-        raise NotImplementedError()
-
-    def _init_image(self,
-                    source_adapter_name: str) -> None:
-
-        logger.info('Initializing target container...')
-        self.container, self.passive_container = self.shdocker.startup(
-            self,
-            source_adapter_name,
-            self.target_arch,
-            self._build_snowshu_envars(
-                self.DOCKER_SNOWSHU_ENVARS))
-
-        logger.info('Container initialized.')
-        while not self.target_database_is_ready():
-            sleep(.5)
-
-        self._initialize_snowshu_meta_database()
-
-    def target_database_is_ready(self) -> bool:
-        return self.container.exec_run(
-            self.DOCKER_READY_COMMAND).exit_code == 0
-
-    def finalize_replica(self) -> None:
-        """ Converts all containers to respective replicas,
-            creates 'latest' if any on the running containers are of local arch
+        Returns the target data type for a given source data type.
+        This helper method translates source data types to target data types.
         """
-        logger.info('Finalizing target container into replica...')
-        self.shdocker.convert_container_to_replica(self.replica_meta['name'],
-                                                   self.container,
-                                                   self.passive_container)
-        logger.info(f'Finalized replica image {self.replica_meta["name"]}')
-
-    def _generate_credentials(self, host) -> Credentials:
-        return Credentials(host=host,
-                           port=self.DOCKER_TARGET_PORT,
-                           **dict(zip(('user',
-                                       'password',
-                                       'database',
-                                       ),
-                                      ['snowshu' for _ in range(3)])))
-
-    def _build_conn_string_partial(
-            self, dialect: str, database: Optional[str] = None) -> str:
-        database = database if database is not None else self._credentials.database
-        conn_string = (f"{self.dialect}://{self._credentials.user}:{self._credentials.password}"
-                       f"@{self._credentials.host}:{self.DOCKER_TARGET_PORT}/{database}?")
-        return conn_string, {USER, PASSWORD, HOST, PORT, DATABASE, }
-
-    def _get_data_type(self, source_type: str) -> DataType:
         try:
-            return self.DATA_TYPE_MAPPINGS[source_type.replace(' ', '_').lower()]
+            return self.DATA_TYPE_MAPPINGS.get(source_type.replace(" ", "_").lower())
```

Comment:
Switching `.get` here will end up returning `None` instead of erroring out as intended.